beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/handler/MessageEventHandler.java

```diff
@@ -0,0 +1,160 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.apiary.handler;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+import com.expedia.apiary.extensions.receiver.common.messaging.MessageEvent;
+
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
+import com.expediagroup.beekeeper.core.model.HousekeepingPath;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.core.model.PathStatus;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.FilterType;
+import com.expediagroup.beekeeper.scheduler.apiary.filter.ListenerEventFilter;
+import com.expediagroup.beekeeper.scheduler.apiary.model.EventModel;
+
+@Component
+public abstract class MessageEventHandler {
+
+  private static final Logger log = LoggerFactory.getLogger(MessageEventHandler.class);
+  private final String CLIENT_ID = "apiary-metastore-event";
+  private final LifecycleEventType lifecycleEventType;
+  private final String cleanupDelay;
+  private final String hivePropertyKey;
+  private final List<FilterType> validFilterClasses;
+  @Autowired private EnumMap<FilterType, ListenerEventFilter> filterTypeMap;
+
+  MessageEventHandler(
+      String cleanupDelay,
+      String hivePropertyKey,
+      LifecycleEventType lifecycleEventType,
+      List<FilterType> validFilterClasses
+  ) {
+    this.cleanupDelay = cleanupDelay;
+    this.hivePropertyKey = hivePropertyKey;
+    this.lifecycleEventType = lifecycleEventType;
+    this.validFilterClasses = validFilterClasses;
+  }
+
+  public List<HousekeepingPath> handleMessage(MessageEvent event) {
+    ListenerEvent listenerEvent = event.getEvent();
+
+    if (shouldFilterMessage(listenerEvent)) {
+      return Collections.emptyList();
+    }
+
+    return generateHouseKeepingPaths(listenerEvent);
+  }
+
+  /**
+   * Get's a boolean flag if the given tableparameters exist for this mapper
+   *
+   * @param tableParameters
+   * @return boolean
+   */
+  boolean checkIfTablePropertyExists(Map<String, String> tableParameters) {
```

Comment:
It looks like this method is duplicating what the `TableParameterListenerEventFilter` class does?

Reply:
yeah - good catch. This is another remnant of the previous PR. Removed.

It also brings up a design question: Should filters have context into what lifecycle event type they are filtering?

Reply:
For now at least, I think not. Whatever filters are injected in to each path handler are the ones that will be used, the reverse relationship doesn't need to be known and I can't think of a usecase for this in the future either. Do you have something in mind?

Reply:
Yeah, specifically for `TableParameterListenerEventFilter` the table param it checks is directly related to the lifecycle type from what I can tell. Given that, if we're using the enum for it, we'd have to pass that in directly from the filter call which is basically how I refactored it again from these PR comments.